packages/edit-site/src/components/dataviews/bulk-actions.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	ToolbarButton,
+	Toolbar,
+	ToolbarGroup,
+	Popover,
+} from '@wordpress/components';
+import { useMemo } from '@wordpress/element';
+import { __, _n, sprintf } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { ActionWithModal } from './item-actions';
+
+function PrimaryActionTrigger( { action, onClick } ) {
+	return (
+		<ToolbarButton
+			label={ action.label }
+			icon={ action.icon }
+			isDestructive={ action.isDestructive }
+			size="compact"
+			onClick={ onClick }
+		/>
+	);
+}
+
+const EMPTY_ARRAY = [];
+
+export default function BulkActions( {
+	data,
+	selection,
+	actions = EMPTY_ARRAY,
+	setSelection,
+} ) {
+	const items = useMemo(
+		() =>
+			data?.filter( ( item ) => selection?.includes( item.id ) ) ??
+			EMPTY_ARRAY,
+		[ data, selection ]
+	);
+	const primaryActions = useMemo(
+		() =>
+			actions.filter( ( action ) => {
+				return (
+					action.isBulk &&
+					action.isPrimary &&
+					items.every( ( item ) => action.isEligible( item ) )
+				);
+			} ),
+		[ actions, items ]
+	);
+
+	if (
+		( selection && selection.length === 0 ) ||
+		primaryActions.length === 0
+	) {
+		return null;
+	}
+
+	return (
+		<Popover
+			placement="top-middle"
+			className="dataviews-bulk-actions-popover"
+		>
+			<Toolbar label="Bulk actions">
+				<div className="dataviews-bulk-actions-toolbar-wrapper">
+					<ToolbarGroup>
+						<ToolbarButton onClick={ () => {} } disabled={ true }>
```

Comment:
I'm not familiar with the toolbar component, so I may be missing context: I found it weird that we use an “inactive” button. Can we use a `ToolbarItem` instead?